### DISCUSSION TITLE
dynamic route default value

### DISCUSSION THREAD

**Discussion by m235** [vercel/next.js#16106] [Help]
Hello,

We currently have a dynamic route set like that `/something/[name]` but we have a 404 when we try to reach /something. Which is totally normal I guess. What is the best way in next for having a default value in a dynamic route ?

**Comment by tinywinds** [comment]
This problem can be solved with nginx's rewrite. The default value is contrary to not-found.js.

**Comment by hwsdien** [comment]
If you use `[slug]` or `[...slug]` in Next.js without providing a `slug`, the route may not render. To address this, you can change `[slug]` to `[[...slug]]`, which ensures that the route matches even when `slug` is not provided. In the code, you can check for the presence of `slug` using `!params.slug ? default_value : params.slug[0]`. It's important to note that this method will capture all subsequent route segments, so you can handle the logic accordingly by checking if the `params.slug` array length is equal to 1.
